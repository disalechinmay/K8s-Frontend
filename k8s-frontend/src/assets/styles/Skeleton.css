body {
	background-image: url("../images/oriental-tiles.png");
	background-repeat: repeat;
	font-family: "Ubuntu", sans-serif;
	font-size: 1.2vw;
}

.mainContainer {
	position: absolute;
	margin: auto;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	width: 100%;
	height: 100%;
	/*border-radius: 10px;*/
	display: flex;
	flex-wrap: wrap;
	box-shadow: 0 8px 10px 0 rgba(0, 0, 0, 0.3), 0 6px 20px 0 rgba(0, 0, 0, 0.3),
		0 8px 10px 0 rgba(0, 0, 0, 0.3), 0 8px 10px 0 rgba(0, 0, 0, 0.3);
}

.sidebar {
	background-color: rgba(42, 42, 42, 0.95); /*#2a2a2a;*/
	min-width: 20%;
	/*border-radius: 10px 0px 0px 10px;*/
	display: flex;
	flex-direction: column;
	flex: 1;
	/*max-height: 100%;*/
	/*overflow: auto;*/
}

.mainContent {
	background-color: rgba(38, 28, 26, 0.9); /*#261c1a;*/
	min-width: 80%;
	/*border-radius: 0px 10px 10px 0px;*/
	display: flex;
	flex-direction: column;
	flex: 2;
	min-height: 100%;
	max-height: 100%;
	overflow: scroll;
}

.sidebar-title {
	font-family: "Oleo Script", cursive;
	color: #dd5135;
	background: -webkit-linear-gradient(#dd5135, #ca4853);
	-webkit-background-clip: text;
	-webkit-text-fill-color: transparent;
	font-size: 3vw;
	font-weight: bold;
	align-self: center;
	margin: 5% 5% 5% 5%;
	cursor: pointer;
	transition: 0.25s;
}

.sidebar-title:hover {
	text-shadow: 1px 0px #7f8c8d;
	transition: 0.25s;
}

.sidebar-namespace-picker {
	background-color: #212322;
	align-self: center;
	width: 100%;
	margin: 5% 1% 5% 1%;
	display: flex;
	flex-direction: column;
}

.sidebar-namespace-picker .namespace-picker-title {
	color: #bdc3c7;
	font-size: 1.2vw;
	font-variant-caps: all-petite-caps;
	align-self: center;
	justify-content: center;
	margin: 1% 1% 3% 1%;
}

.sidebar-namespace-picker .namespace-picker {
	color: #656565;
	align-self: center;
	justify-content: center;
	margin: 2% 1% 2% 1%;
}

.sidebar-namespace-picker .namespace-picker select {
	background-color: #212322;
	text-align-last: center;
	color: #2ecc71;
	width: 100%;
	border: none;
	font-size: 1.2vw;
	-webkit-appearance: button;
	appearance: button;
	outline: none;
}

.sidebar-section-title {
	/*text-align: center;*/
	margin: 5% 1% 3% 10%;
	color: #ecf0f1;
	width: 100%;
	font-weight: bolder;
}

.sidebar-button {
	align-content: center;
	color: #ecf0f1;
	cursor: pointer;
	padding: 5% 10% 5% 10%;
	transition: 0.25s;
}

.sidebar-button:hover {
	background-color: #212322;
	color: white;
	transition: 0.25s;
}

.sidebar-button-active {
	align-content: center;
	background-color: rgba(42, 42, 42, 0.95);
	box-shadow: 0 5px 5px 0 rgba(0, 0, 0, 0.3);
	border-radius: 0px 100px 100px 0px;
	color: #ecf0f1;
	cursor: pointer;
	padding: 5% 10% 5% 10%;
	border-right: 4px solid #dd5135;
	transition: 0.25s;
}

.sidebar-search {
	width: 80%;
	margin: 5% 5% 5% 5%;
	padding: 3% 3% 3% 3%;
	background-color: rgba(42, 42, 42, 0.95);
	border-radius: 5px 5px 5px 5px;
	box-shadow: 0 5px 5px 0 rgba(0, 0, 0, 0.3);
	border-bottom: 1px solid #0f4626;
	color: #bdc3c7;
	transition: 0.25s;
}

.sidebar-search:hover {
	outline: none;
	border-bottom: 1px solid #2ecc71;
	transition: 0.25s;
}

.sidebar-search:focus {
	color: #ecf0f1;
	outline: none;
	border-bottom: 1px solid #2ecc71;
	transition: 0.25s;
}

@media only screen and (max-width: 1000px) {
	.mainContainer {
		width: 100%;
		height: 100%;
		border-radius: 0px;
		flex-direction: column;
		flex-wrap: nowrap;
	}

	.sidebar {
		border-radius: 0px 0px 0px 0px;
		align-self: flex-start;
		min-width: 99%;
	}

	.mainContent {
		border-radius: 0px 0px 0px 0px;
		margin-top: 1%;
		align-self: flex-start;
		min-width: 99%;
		max-width: 99%;
		overflow: all;
	}

	.sidebar-title {
		margin: 1% 1% 1% 1%;
	}

	.sidebar-namespace-picker {
		margin: 1% 1% 1% 1%;
	}

	.sidebar-namespace-picker .namespace-picker {
		margin: 1% 1% 1% 1%;
	}
	.sidebar-namespace-picker .namespace-picker-title {
		margin: 1% 1% 1% 1%;
	}

	.sidebar-section-title {
		margin: 1% 1% 1% 1%;
	}

	.sidebar-button {
		padding: 2% 10% 1% 10%;
	}

	.sidebar-button-active {
		border-radius: 100px 100px 100px 100px;
		border-left: 4px solid #dd5135;
	}
}

@media only screen and (max-height: 600px) {
	.mainContainer {
		width: 100%;
		height: 100%;
		border-radius: 0px;
		flex-direction: column;
		flex-wrap: nowrap;
	}

	.sidebar {
		border-radius: 0px 0px 0px 0px;
		align-self: flex-start;
		min-width: 99%;
	}

	.mainContent {
		border-radius: 0px 0px 0px 0px;
		margin-top: 1%;
		align-self: flex-start;
		min-width: 99%;
		max-width: 99%;
		overflow: all;
	}

	.sidebar-title {
		margin: 1% 1% 1% 1%;
	}

	.sidebar-namespace-picker {
		margin: 1% 1% 1% 1%;
	}

	.sidebar-namespace-picker .namespace-picker {
		margin: 1% 1% 1% 1%;
	}
	.sidebar-namespace-picker .namespace-picker-title {
		margin: 1% 1% 1% 1%;
	}

	.sidebar-section-title {
		margin: 1% 1% 1% 1%;
	}

	.sidebar-button {
		padding: 2% 10% 1% 10%;
	}

	.sidebar-button-active {
		border-radius: 100px 100px 100px 100px;
		border-left: 4px solid #dd5135;
	}
}

.animated {
	transition: 0.25s;
}

/* width */
::-webkit-scrollbar {
	height: 0px;
	width: 0px;
	visibility: hidden;
	transition: 0.25s;
	border-radius: 5px 5px 5px 5px;
}

/* Track */
::-webkit-scrollbar-track {
	background: transparent;
	transition: 0.25s;
	border-radius: 10px;
}

/* Handle */
::-webkit-scrollbar-thumb {
	background: #dd5135;
	transition: 0.25s;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
	background: #dd5135;
	transition: 0.25s;
}

.fa {
	width: 20px;
}
