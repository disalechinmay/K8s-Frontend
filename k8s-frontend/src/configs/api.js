export const API_LOCATION = "http://localhost:5000";
