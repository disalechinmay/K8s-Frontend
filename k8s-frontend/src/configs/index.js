export { API_LOCATION } from "./api";
export { SIDEBAR_OPTIONS, PAGE_LOAD_LIMIT_TIME } from "./constants";
