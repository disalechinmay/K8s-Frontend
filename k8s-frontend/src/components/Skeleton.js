import React, { Component } from "react";
import { getNamespaces } from "../services";
import { LoadingPage, ErrorPage } from "./common";
import { HomePage } from "./HomePage";
import { NodesPage } from "./NodesPage";
import { PodsPage } from "./PodsPage";
import { PodAddPage } from "./PodAddPage";
import { PodEditPage } from "./PodEditPage";
import { DeploymentsPage } from "./DeploymentsPage";
import { ConfigMapsPage } from "./ConfigMapsPage";
import { ConfigMapAddPage } from "./ConfigMapAddPage";
import { ConfigMapEditPage } from "./ConfigMapEditPage";
import { DeploymentEditPage } from "./DeploymentEditPage";
import { ServicesPage } from "./ServicesPage";
import { ServiceEditPage } from "./ServiceEditPage";
import { JobsPage } from "./JobsPage";
import { SecretsPage } from "./SecretsPage";
import { SecretAddPage } from "./SecretAddPage";
import { SecretEditPage } from "./SecretEditPage";
import { SearchBar, SearchPage } from "./SearchPage";
import { SIDEBAR_OPTIONS } from "../configs";

class Skeleton extends Component {
	state = {
		sidebarOptionSelected: 0,
		pageLoading: true,
		namespacesListSet: false,
		namespacesList: [],
		namespaceSelected: "default",
		errorSet: false,
		errorDescription: "",
		searchTokens: [],
		editResourceName: "" // Used as a prop to Edit Resource Page
	};

	// Creates sidebar button references.
	// Sets namespacesList using a service call.
	constructor(props) {
		super(props);

		// List of buttons which will appear on sidebar.
		this.sidebarOptions = SIDEBAR_OPTIONS;

		// References for sidebar buttons.
		this.optionRefs = [];
		for (let iter = 0; iter < this.sidebarOptions.length; iter++)
			this.optionRefs[iter] = React.createRef();

		// Fetching list of namespaces from backend.
		getNamespaces()
			.then(result => {
				this.setState({
					...this.state,
					pageLoading: false,
					namespacesListSet: true,
					namespacesList: result.payLoad
				});
			})
			.catch(error => {
				this.setState({
					...this.state,
					errorSet: true,
					errorDescription: error
				});
			});
	}

	// Updates the state & style when sidebar options are selected.
	handleClick(event, clickedOptionIndex) {
		let newState = { ...this.state };

		for (let iter = 0; iter < this.sidebarOptions.length; iter++)
			if (
				this.optionRefs[iter].current.classList.contains(
					"sidebar-button-active"
				)
			)
				this.optionRefs[iter].current.classList.remove(
					"sidebar-button-active"
				);

		newState.sidebarOptionSelected = clickedOptionIndex;

		if (clickedOptionIndex !== 0)
			event.target.classList.add("sidebar-button-active");

		this.setState(newState);
	}

	// Updates the state when namespace selector changes.
	handleNamespaceChange(event) {
		this.setState({ ...this.state, namespaceSelected: event.target.value });
	}

	// Sets sidebarOptionSelected to 9, which in turn renders SearchPage.
	// Manipulated by SearchBar.
	renderSearchPage() {
		for (let iter = 0; iter < this.sidebarOptions.length; iter++)
			if (
				this.optionRefs[iter].current.classList.contains(
					"sidebar-button-active"
				)
			)
				this.optionRefs[iter].current.classList.remove(
					"sidebar-button-active"
				);
		this.setState({ ...this.state, sidebarOptionSelected: 9 });
	}

	// Sets searchTokens in state, which is in turn sent to SearchPage.
	sendTokens(tokens) {
		this.setState({
			...this.state,
			searchTokens: tokens,
			sidebarOptionSelected: 9
		});
	}

	// Sets editResourceName and renders the resource edit page
	renderResourceEditPage(resourceType, resourceName) {
		if (resourceType === "DEPLOYMENT") {
			this.setState({
				...this.state,
				editResourceName: resourceName,
				sidebarOptionSelected: 13
			});
		} else if (resourceType === "SECRET") {
			this.setState({
				...this.state,
				editResourceName: resourceName,
				sidebarOptionSelected: 18
			});
		} else if (resourceType === "POD") {
			this.setState({
				...this.state,
				editResourceName: resourceName,
				sidebarOptionSelected: 12
			});
		} else if (resourceType === "CONFIG_MAP") {
			this.setState({
				...this.state,
				editResourceName: resourceName,
				sidebarOptionSelected: 17
			});
		} else if (resourceType === "SERVICE") {
			this.setState({
				...this.state,
				editResourceName: resourceName,
				sidebarOptionSelected: 14
			});
		}
	}

	// Renders resource add page
	renderResourceAddPage(resourceType) {
		if (resourceType === "SECRET") {
			this.setState({
				...this.state,
				sidebarOptionSelected: 28
			});
		} else if (resourceType === "CONFIG_MAP") {
			this.setState({
				...this.state,
				sidebarOptionSelected: 27
			});
		} else if (resourceType === "POD") {
			this.setState({
				...this.state,
				sidebarOptionSelected: 23
			});
		}
	}

	render() {
		// Renders ErrorPage if errorSet is true.
		if (this.state.errorSet)
			return <ErrorPage errorDescription={this.state.errorDescription} />;

		// Renders LoadingPage if pageLoading is true.
		if (this.state.pageLoading) return <LoadingPage />;

		return (
			<React.Fragment>
				{/* Main Container */}
				<div className="mainContainer">
					{/* Sidebar */}
					<div className="sidebar">
						{/* Sidebar Title*/}
						<div
							className="sidebar-title"
							onClick={event => this.handleClick(event, 0)}
						>
							Symphonize
						</div>

						{/* Sidebar Namespace Picker*/}
						<div className="sidebar-namespace-picker">
							<span className="flex flex-row">
								<span className="flex flex-column justify-content-center namespace-settings">
									<span className="fa fa-cog">
										<span className="menu">
											<span onClick="">Add</span>
											<span>Delete</span>
										</span>
									</span>
								</span>

								<div className="namespace-picker">
									<select
										className="namespace-selector"
										onChange={event =>
											this.handleNamespaceChange(event)
										}
									>
										{/* Populates options of namespaces */}
										{this.state.namespacesListSet &&
											this.state.namespacesList.map(
												namespace => {
													return (
														<option
															value={namespace}
															key={
																namespace +
																"_NAMESPACE"
															}
														>
															{namespace}
														</option>
													);
												}
											)}
									</select>
									<div className="namespace-picker-title">
										Namespace
									</div>
								</div>
							</span>
						</div>

						{/* Sidebar Buttons*/}
						<div className="sidebar-section-title">Resources</div>
						{/* Rendering sidebar buttons */}
						{this.sidebarOptions.map((sidebarOption, index) => {
							return (
								<div
									className="sidebar-button"
									id={sidebarOption.id}
									key={index + "_SIDEBAR_BUTTON"}
									onClick={event =>
										this.handleClick(event, index + 1)
									}
									ref={this.optionRefs[index]}
								>
									<span
										onClick={null}
										className={sidebarOption.style}
									></span>
									&emsp; {sidebarOption.title}
								</div>
							);
						})}

						{/* Sidebar SearchBar*/}
						<SearchBar
							renderSearchPage={() => this.renderSearchPage()}
							sendTokens={tokens => this.sendTokens(tokens)}
						/>
					</div>

					{/* Main Content: Displays core pages.*/}
					<div className="mainContent" id="mainContent">
						{/* Depending upon sidebarOptionSelected, render respective page. */}

						{this.state.sidebarOptionSelected === 0 && (
							<HomePage
								namespace={this.state.namespaceSelected}
							/>
						)}

						{this.state.sidebarOptionSelected === 1 && (
							<NodesPage />
						)}

						{this.state.sidebarOptionSelected === 2 && (
							<PodsPage
								namespace={this.state.namespaceSelected}
								renderAddPage={resourceType =>
									this.renderResourceAddPage(resourceType)
								}
								renderEditPage={(resourceType, resourceName) =>
									this.renderResourceEditPage(
										resourceType,
										resourceName
									)
								}
							/>
						)}

						{this.state.sidebarOptionSelected === 3 && (
							<DeploymentsPage
								renderEditPage={(resourceType, resourceName) =>
									this.renderResourceEditPage(
										resourceType,
										resourceName
									)
								}
								namespace={this.state.namespaceSelected}
							/>
						)}

						{this.state.sidebarOptionSelected === 4 && (
							<ServicesPage
								renderEditPage={(resourceType, resourceName) =>
									this.renderResourceEditPage(
										resourceType,
										resourceName
									)
								}
								namespace={this.state.namespaceSelected}
							/>
						)}
						{this.state.sidebarOptionSelected === 5 && (
							<JobsPage
								namespace={this.state.namespaceSelected}
							/>
						)}

						{this.state.sidebarOptionSelected === 7 && (
							<ConfigMapsPage
								renderEditPage={(resourceType, resourceName) =>
									this.renderResourceEditPage(
										resourceType,
										resourceName
									)
								}
								namespace={this.state.namespaceSelected}
								renderAddPage={resourceType =>
									this.renderResourceAddPage(resourceType)
								}
							/>
						)}

						{this.state.sidebarOptionSelected === 8 && (
							<SecretsPage
								renderEditPage={(resourceType, resourceName) =>
									this.renderResourceEditPage(
										resourceType,
										resourceName
									)
								}
								renderAddPage={resourceType =>
									this.renderResourceAddPage(resourceType)
								}
								namespace={this.state.namespaceSelected}
							/>
						)}

						{this.state.sidebarOptionSelected === 9 && (
							<SearchPage
								namespace={this.state.namespaceSelected}
								searchTokens={this.state.searchTokens}
							/>
						)}

						{this.state.sidebarOptionSelected === 12 && (
							<PodEditPage
								namespace={this.state.namespaceSelected}
								resourceName={this.state.editResourceName}
							/>
						)}

						{this.state.sidebarOptionSelected === 13 && (
							<DeploymentEditPage
								namespace={this.state.namespaceSelected}
								resourceName={this.state.editResourceName}
							/>
						)}

						{this.state.sidebarOptionSelected === 14 && (
							<ServiceEditPage
								namespace={this.state.namespaceSelected}
								resourceName={this.state.editResourceName}
							/>
						)}
						{this.state.sidebarOptionSelected === 17 && (
							<ConfigMapEditPage
								namespace={this.state.namespaceSelected}
								resourceName={this.state.editResourceName}
							/>
						)}

						{this.state.sidebarOptionSelected === 18 && (
							<SecretEditPage
								namespace={this.state.namespaceSelected}
								resourceName={this.state.editResourceName}
							/>
						)}

						{this.state.sidebarOptionSelected === 23 && (
							<PodAddPage
								namespace={this.state.namespaceSelected}
							/>
						)}
						{this.state.sidebarOptionSelected === 27 && (
							<ConfigMapAddPage
								namespace={this.state.namespaceSelected}
							/>
						)}
						{this.state.sidebarOptionSelected === 28 && (
							<SecretAddPage
								namespace={this.state.namespaceSelected}
							/>
						)}
					</div>
				</div>
			</React.Fragment>
		);
	}
}

export default Skeleton;
