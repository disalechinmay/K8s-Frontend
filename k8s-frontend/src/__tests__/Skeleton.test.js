import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";
import { PAGE_LOAD_LIMIT_TIME } from "../configs";
import Skeleton from "../components/Skeleton.js";

describe("Skeleton Tests", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(<Skeleton />);
	});

	test("Check if loading page is being rendered initially.", () => {
		expect(wrapper.find("LoadingPage").exists()).toBeTruthy();
	});

	test("Check if sidebar title contains text 'Symphonize'", async () => {
		// Waits for 2 second(s) and then updates the wrapper
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		expect(wrapper.find(".sidebar-title").exists()).toBeTruthy();
		expect(wrapper.find(".sidebar-title").text()).toBe("Symphonize");
	});

	test("Check if clicks on sidebar buttons change state appropriately.", async () => {
		// Waits for PAGE_LOAD_LIMIT_TIME second(s) and then updates the wrapper
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		expect(wrapper.state("sidebarOptionSelected")).toBe(0);

		let buttonsToCheck = [
			{ id: "#nodes_sidebarOption", value: 1 },
			{ id: "#pods_sidebarOption", value: 2 },
			{ id: "#deployments_sidebarOption", value: 3 },
			{ id: "#services_sidebarOption", value: 4 },
			{ id: "#jobs_sidebarOption", value: 5 },
			{ id: "#cronJobs_sidebarOption", value: 6 },
			{ id: "#configMaps_sidebarOption", value: 7 },
			{ id: "#secrets_sidebarOption", value: 8 }
		];

		for (let button of buttonsToCheck) {
			wrapper.find(button.id).simulate("click");
			wrapper.update();
			expect(wrapper.state("sidebarOptionSelected")).toBe(button.value);
		}
	});

	test("Check if namespace selector works.", async () => {
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		expect(wrapper.state("namespaceSelected")).toBe("default");
		wrapper
			.find(".namespace-selector")
			.at(0)
			.simulate("change", { target: { value: "a" } });
		expect(wrapper.state("namespaceSelected")).toBe("a");
	});

	test("Check if renderSearchPage() works.", async () => {
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		wrapper.instance().renderSearchPage();
		expect(wrapper.state("sidebarOptionSelected")).toBe(9);
	});

	test("Check if sendTokens() works.", async () => {
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		wrapper.instance().sendTokens(["1", "2"]);
		expect(wrapper.state("searchTokens")).toStrictEqual(["1", "2"]);
	});

	test("Check if ErrorPage renders when errorSet is true.", async () => {
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		wrapper.setState({
			...wrapper.state(),
			errorSet: true,
			errorDescription: {
				errorDescription:
					"Something went wrong while retrieving namespaces!",
				errorSuggestions: [
					"Make sure the backend service is up and running.",
					"Make sure the endpoint being accessed is valid."
				]
			}
		});

		wrapper.update();
		expect(wrapper.find("ErrorPage").exists()).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	test("Check if loading page renders correctly.", () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	test("Check if skeleton renders correctly.", async () => {
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
