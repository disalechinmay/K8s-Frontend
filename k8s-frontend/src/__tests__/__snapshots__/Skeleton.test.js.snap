// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Tests Check if ErrorPage renders when errorSet is true. 1`] = `
<Skeleton>
  <ErrorPage
    errorDescription={
      Object {
        "errorDescription": "Something went wrong while retrieving namespaces!",
        "errorSuggestions": Array [
          "Make sure the backend service is up and running.",
          "Make sure the endpoint being accessed is valid.",
        ],
      }
    }
  >
    <div
      className="error-page"
    >
      <div
        className="error-page-content"
      >
        <h1
          className="oops"
        >
          OOPS!
        </h1>
        <span
          className="message"
        >
          Things seem to be broken a bit.
        </span>
        <br />
        <br />
        <span
          className="font-weight-bold"
        >
          Error: 
        </span>
        <span>
          Something went wrong while retrieving namespaces!
        </span>
        <br />
        <br />
        <span
          className="font-weight-bold"
        >
          Suggestions: 
        </span>
        <br />
        <span>
          Make sure the backend service is up and running.
          <br
            key="0_SUGGESTION"
          />
          Make sure the endpoint being accessed is valid.
          <br
            key="1_SUGGESTION"
          />
        </span>
        <span
          className="logo"
        >
          Symphonize
        </span>
      </div>
    </div>
  </ErrorPage>
</Skeleton>
`;

exports[`Skeleton Tests Check if loading page renders correctly. 1`] = `
<Skeleton>
  <LoadingPage>
    <div
      className="loading-page"
    >
      <div
        className="loading-page-content"
      >
        <span
          className="loader"
        >
          <Loader
            color="#ecf0f1"
            css=""
            loading={true}
            size="5em"
          >
            <EmotionCssPropInternal
              __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
              css={
                Array [
                  Object {
                    "map": undefined,
                    "name": "1oyc5ch",
                    "next": undefined,
                    "styles": "
      position: relative;
      width: 5em;
      height: 5em;
    ",
                    "toString": [Function],
                  },
                  "",
                ]
              }
            >
              <div
                className="css-8ytart"
              >
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
                  css={
                    Object {
                      "map": undefined,
                      "name": "1vpk80i",
                      "next": Object {
                        "name": "animation-zodxwl",
                        "next": undefined,
                        "styles": "@keyframes animation-zodxwl{
  0% {transform: rotate(0deg)}
  50% {transform: rotate(180deg)}
  100% {transform: rotate(360deg)}
}",
                      },
                      "styles": "
      position: absolute;
      height: 5em;
      width: 5em;
      border: 1px solid #ecf0f1;
      border-radius: 100%;
      transition: 2s;
      border-bottom: none;
      border-right: none;
      top: 0%;
      left: 0%;
      animation-fill-mode: \\"\\";
      animation: animation-zodxwl 1s 0s infinite linear;
    ",
                      "toString": [Function],
                    }
                  }
                >
                  <div
                    className="css-1vpk80i"
                  />
                </EmotionCssPropInternal>
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
                  css={
                    Object {
                      "map": undefined,
                      "name": "pwsp8i",
                      "next": Object {
                        "name": "animation-zodxwl",
                        "next": undefined,
                        "styles": "@keyframes animation-zodxwl{
  0% {transform: rotate(0deg)}
  50% {transform: rotate(180deg)}
  100% {transform: rotate(360deg)}
}",
                      },
                      "styles": "
      position: absolute;
      height: 4.5em;
      width: 4.5em;
      border: 1px solid #ecf0f1;
      border-radius: 100%;
      transition: 2s;
      border-bottom: none;
      border-right: none;
      top: 1.75%;
      left: 0.875%;
      animation-fill-mode: \\"\\";
      animation: animation-zodxwl 1s 0.2s infinite linear;
    ",
                      "toString": [Function],
                    }
                  }
                >
                  <div
                    className="css-pwsp8i"
                  />
                </EmotionCssPropInternal>
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
                  css={
                    Object {
                      "map": undefined,
                      "name": "1lcb3ax",
                      "next": Object {
                        "name": "animation-zodxwl",
                        "next": undefined,
                        "styles": "@keyframes animation-zodxwl{
  0% {transform: rotate(0deg)}
  50% {transform: rotate(180deg)}
  100% {transform: rotate(360deg)}
}",
                      },
                      "styles": "
      position: absolute;
      height: 4em;
      width: 4em;
      border: 1px solid #ecf0f1;
      border-radius: 100%;
      transition: 2s;
      border-bottom: none;
      border-right: none;
      top: 3.5%;
      left: 1.75%;
      animation-fill-mode: \\"\\";
      animation: animation-zodxwl 1s 0.4s infinite linear;
    ",
                      "toString": [Function],
                    }
                  }
                >
                  <div
                    className="css-1lcb3ax"
                  />
                </EmotionCssPropInternal>
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
                  css={
                    Object {
                      "map": undefined,
                      "name": "8qzkbb",
                      "next": Object {
                        "name": "animation-zodxwl",
                        "next": undefined,
                        "styles": "@keyframes animation-zodxwl{
  0% {transform: rotate(0deg)}
  50% {transform: rotate(180deg)}
  100% {transform: rotate(360deg)}
}",
                      },
                      "styles": "
      position: absolute;
      height: 3.5em;
      width: 3.5em;
      border: 1px solid #ecf0f1;
      border-radius: 100%;
      transition: 2s;
      border-bottom: none;
      border-right: none;
      top: 5.249999999999999%;
      left: 2.6249999999999996%;
      animation-fill-mode: \\"\\";
      animation: animation-zodxwl 1s 0.6000000000000001s infinite linear;
    ",
                      "toString": [Function],
                    }
                  }
                >
                  <div
                    className="css-8qzkbb"
                  />
                </EmotionCssPropInternal>
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
                  css={
                    Object {
                      "map": undefined,
                      "name": "1ewkqe9",
                      "next": Object {
                        "name": "animation-zodxwl",
                        "next": undefined,
                        "styles": "@keyframes animation-zodxwl{
  0% {transform: rotate(0deg)}
  50% {transform: rotate(180deg)}
  100% {transform: rotate(360deg)}
}",
                      },
                      "styles": "
      position: absolute;
      height: 3em;
      width: 3em;
      border: 1px solid #ecf0f1;
      border-radius: 100%;
      transition: 2s;
      border-bottom: none;
      border-right: none;
      top: 7%;
      left: 3.5%;
      animation-fill-mode: \\"\\";
      animation: animation-zodxwl 1s 0.8s infinite linear;
    ",
                      "toString": [Function],
                    }
                  }
                >
                  <div
                    className="css-1ewkqe9"
                  />
                </EmotionCssPropInternal>
              </div>
            </EmotionCssPropInternal>
          </Loader>
        </span>
        <span
          className="message"
        >
          Loading
        </span>
        <span
          className="logo"
        >
          Symphonize
        </span>
      </div>
    </div>
  </LoadingPage>
</Skeleton>
`;

exports[`Skeleton Tests Check if skeleton renders correctly. 1`] = `
<Skeleton>
  <div
    className="mainContainer"
  >
    <div
      className="sidebar"
    >
      <div
        className="sidebar-title"
        onClick={[Function]}
      >
        Symphonize
      </div>
      <div
        className="sidebar-namespace-picker"
      >
        <div
          className="namespace-picker"
        >
          <select
            className="namespace-selector"
            onChange={[Function]}
          >
            <option
              key="default_NAMESPACE"
              value="default"
            >
              default
            </option>
            <option
              key="docker_NAMESPACE"
              value="docker"
            >
              docker
            </option>
            <option
              key="kube-node-lease_NAMESPACE"
              value="kube-node-lease"
            >
              kube-node-lease
            </option>
            <option
              key="kube-public_NAMESPACE"
              value="kube-public"
            >
              kube-public
            </option>
            <option
              key="kube-system_NAMESPACE"
              value="kube-system"
            >
              kube-system
            </option>
            <option
              key="kubernetes-dashboard_NAMESPACE"
              value="kubernetes-dashboard"
            >
              kubernetes-dashboard
            </option>
          </select>
        </div>
        <div
          className="namespace-picker-title"
        >
          Namespace
        </div>
      </div>
      <div
        className="sidebar-section-title"
      >
        Resources
      </div>
      <div
        className="sidebar-button"
        id="nodes_sidebarOption"
        key="0_SIDEBAR_BUTTON"
        onClick={[Function]}
      >
        <span
          className="fa fa-desktop"
        />
          
        Nodes
      </div>
      <div
        className="sidebar-button"
        id="pods_sidebarOption"
        key="1_SIDEBAR_BUTTON"
        onClick={[Function]}
      >
        <span
          className="fa fa-archive"
        />
          
        Pods
      </div>
      <div
        className="sidebar-button"
        id="deployments_sidebarOption"
        key="2_SIDEBAR_BUTTON"
        onClick={[Function]}
      >
        <span
          className="fa fa-cubes"
        />
          
        Deployments
      </div>
      <div
        className="sidebar-button"
        id="services_sidebarOption"
        key="3_SIDEBAR_BUTTON"
        onClick={[Function]}
      >
        <span
          className="fa fa-random"
        />
          
        Services
      </div>
      <div
        className="sidebar-button"
        id="jobs_sidebarOption"
        key="4_SIDEBAR_BUTTON"
        onClick={[Function]}
      >
        <span
          className="fa fa-tasks"
        />
          
        Jobs
      </div>
      <div
        className="sidebar-button"
        id="cronJobs_sidebarOption"
        key="5_SIDEBAR_BUTTON"
        onClick={[Function]}
      >
        <span
          className="fa fa-clock-o"
        />
          
        Cron Jobs
      </div>
      <div
        className="sidebar-button"
        id="configMaps_sidebarOption"
        key="6_SIDEBAR_BUTTON"
        onClick={[Function]}
      >
        <span
          className="fa fa-gears"
        />
          
        Config Maps
      </div>
      <div
        className="sidebar-button"
        id="secrets_sidebarOption"
        key="7_SIDEBAR_BUTTON"
        onClick={[Function]}
      >
        <span
          className="fa fa-user-secret"
        />
          
        Secrets
      </div>
      <SearchBar
        refreshState={[Function]}
        renderSearchPage={[Function]}
        sendTokens={[Function]}
      >
        <div
          className="sidebar-search"
          contentEditable="true"
          onBlur={[Function]}
          onClick={[Function]}
          onInput={[Function]}
          suppressContentEditableWarning="true"
        >
          Search Resources..
        </div>
      </SearchBar>
    </div>
    <div
      className="mainContent"
      id="mainContent"
    />
  </div>
</Skeleton>
`;
